xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/objects/classes/NumberClass.java

```diff
@@ -126,28 +132,30 @@ public BaseProperty fromString(String value)
         BaseProperty property = newProperty();
         String ntype = getNumberType();
         Number nvalue = null;
-
-        try {
-            if (ntype.equals(TYPE_INTEGER)) {
-                if ((value != null) && (!value.equals(""))) {
-                    nvalue = Integer.valueOf(value);
-                }
-            } else if (ntype.equals(TYPE_FLOAT)) {
-                if ((value != null) && (!value.equals(""))) {
-                    nvalue = Float.valueOf(value);
-                }
-            } else if (ntype.equals(TYPE_DOUBLE)) {
-                if ((value != null) && (!value.equals(""))) {
-                    nvalue = Double.valueOf(value);
-                }
-            } else {
-                if ((value != null) && (!value.equals(""))) {
-                    nvalue = Long.valueOf(value);
+        boolean valueBlank = StringUtils.isBlank(value);
+
+        if (!valueBlank) {
+            try {
+                NumberFormat numberFormat = NumberFormat.getNumberInstance(getOwnerDocument().getLocale());
```

Comment:
Using `#getLocale()` does not really help as the xobject are located on the default document, which is always Local.ROOT.

`#getRealLocale()` should give better results